modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/listk.kt

```diff
@@ -279,3 +291,22 @@ interface ListKAlternative : Alternative<ForListK>, ListKApplicative {
   override fun <A> Kind<ForListK, A>.orElse(b: Kind<ForListK, A>): Kind<ForListK, A> =
     (this.fix() + b.fix()).k()
 }
+
+@extension
+interface ListKSemialign : Semialign<ForListK>, ListKFunctor {
+  override fun <A, B> align(
+    left: Kind<ForListK, A>,
+    right: Kind<ForListK, B>
+  ): Kind<ForListK, Ior<A, B>> {
+    // from arrow-syntax
+    fun <T> List<T>.tail(): List<T> = this.drop(1)
+
+    fun <X, Y> alignRec(ls: List<X>, rs: List<Y>): List<Ior<X, Y>> = when {
+      ls.isEmpty() -> rs.map { it.rightIor() }
+      rs.isEmpty() -> ls.map { it.leftIor() }
+      else -> listOf(Ior.Both(ls.first(), rs.first())).listPlus(alignRec(ls.tail(), rs.tail()))
+    }
+
+    return alignRec(left.fix(), right.fix()).k()
```

Comment:
My initial comment was a bit early, wanted that in a review because I often change my thoughts :D

Either way there is no need for functions in here, although I am surprised tail is not available here. Check again after moving the tests.

Reply:
> Check again after moving the tests.
Ignore that part, I was somewhere else with my thoughts :D
Just use drop(1) instead, its just twice anyway, we have tail as an extension function somewhere tho

Reply:
ok, using drop now directly. the extensions function is in the module arrow-syntax thats not in the dependencies